apps/api/src/app/subscribers/usecases/update-subscriber-channel/update-subscriber-channel.usecase.ts

```diff
@@ -0,0 +1,77 @@
+import { Injectable } from '@nestjs/common';
+import { IDirectChannel, SubscriberRepository, IntegrationRepository } from '@novu/dal';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+import { UpdateSubscriberChannelCommand } from './update-subscriber-channel.command';
+
+@Injectable()
+export class UpdateSubscriberChannel {
+  constructor(
+    private subscriberRepository: SubscriberRepository,
+    private integrationRepository: IntegrationRepository
+  ) {}
+
+  async execute(command: UpdateSubscriberChannelCommand) {
+    const foundSubscriber = await this.subscriberRepository.findBySubscriberId(
+      command.environmentId,
+      command.subscriberId
+    );
+
+    if (!foundSubscriber) {
+      throw new ApiException(`SubscriberId: ${command.subscriberId} not found`);
+    }
+
+    const foundIntegration = await this.integrationRepository.findOne({
+      _environmentId: command.environmentId,
+      providerId: command.integrationId,
```

Comment:
```suggestion
      providerId: command.providerId,
```

Maybe worth calling this providerId to better differentiate from the integration._id